### ISSUE TITLE
Outdated example 

### ISSUE THREAD

**Issue by pMertDogan** [GIfatahTH/states_rebuilder#84]
Hi,
Currently i was looking to [example 8 ](https://github.com/GIfatahTH/states_rebuilder/tree/master/examples/008-clean_architecture_firebase_login) but its outdated.

When i follow example my app is not work as expected.I checked example files and they are not same with example.So i waste time on outdated example.Please update it for others

**Comment by GIfatahTH** [comment]
I am updating it now

**Comment by pMertDogan** [comment]
I have some problems when i apply your example to my own.

I stored email,name and password value in UserVM.I use TabBarView for changing to SingIn and SingUp UI.

When login TabBarButton clicked two times I let the user login/singup.

Problem is when i was filled email and password forms and switched singup UI then go back to singin uı my UserVM values return to empty.


prints from TabBar
```
I/flutter ( 9474): tap index 0  oldIndex 0
I/flutter ( 9474): tap index correct
I/flutter ( 9474): email pmesystem@gmail.com  pass securepassss
I/flutter ( 9474): Lets login user

// If i was switch to singup uı then i was going back to singIn  and clicked login button

I/flutter ( 9474): tap index 0  oldIndex 0
I/flutter ( 9474): tap index correct
I/flutter ( 9474): email null  pass null
```




My Main.dart
```
class UserVM implements AuthBase {
  final AuthBase authService;

  UserVM(this.authService);

  User user;
  String registerLoginEmail;
  String registerLoginPassword;
  String registerLoginName;

  bool fieldsCheck({bool registermode = true}) {
    if (registermode) {
      return registerLoginEmail != null && registerLoginPassword != null;
    } else {
      return registerLoginEmail != null &&
          registerLoginPassword != null &&
          registerLoginName != null;
    }
  }

  @override
  bool passwordReset(String email) {
    // TODO: implement passwordReset
    return null;
  }

  @override
  Future<User> singInWithEmailAndPass(String email, password) async {
    user = await authService.singInWithEmailAndPass(email, password);
    return user;
  }

  @override
  Future<User> singUpWithEmailAndPass(String email, password, name) async {
    user = await authService.singUpWithEmailAndPass(email, password, name);
    print("userVM UserName " + user.userName.toString());
    return user;
  }

  @override
  Future<User> getCurrentUser() async {
    user = await authService.getCurrentUser();
    return user;
  }
}
```

I use tabBar buttons for control sing up and login process

```
class _TabBarUIState extends State<TabBarUI> {
  int lastTapIndex = 0;
  final RMKey loginRMKey = RMKey<String>("Login");
  final RMKey singUpRMKey = RMKey<String>("SingUp button RM Key");

  @override
  Widget build(BuildContext context) {
    return Container(
      height: 50,
      color: UIColors.kapaliMavi,
      child: TabBar(
        onTap: (index) {
          loginRMKey.value = index == 1 ? "I have a account" : "Login";
          singUpRMKey.value = index == 0 ? "Create new account" : "SingUp";

          print("tap index $index  oldIndex $lastTapIndex");
          if (lastTapIndex == index) {
            //Check is button 2x
            print("tap index correct");
            //Try sing in or sing up
            RM.getSetState<UserVM>((s) async {
              switch (index) {
                case (0):
                  print("email " +
                      s.registerLoginEmail.toString() +
                      "  pass " +
                      s.registerLoginPassword.toString());
                  if (s.fieldsCheck()) {
                    print("Lets login user");
                    try {
                      await s.singInWithEmailAndPass(
                          s.registerLoginEmail, s.registerLoginPassword);
                    } on Exception catch (e) {
                      print("login error " + e.toString());
                      Scaffold.of(context).showSnackBar(SnackBar(
                        content: Text(e.toString()),
                      ));
                    }
                  }
                  break;
                case (1):
                  print("Lets singUP user");
                  if (s.fieldsCheck(registermode: true)) {
                    await s.singUpWithEmailAndPass(s.registerLoginEmail,
                        s.registerLoginPassword, s.registerLoginName);
                  }
                  break;
              }
            },
                catchError: true,
                onError: ((context, e) =>
                    Scaffold.of(context).showSnackBar(SnackBar(
                      content: Text(e.toString()),
                    ))));
          }
          lastTapIndex = index;
        },
        indicatorColor: UIColors.grey,
        tabs: [
          StateBuilder<String>(
            rmKey: loginRMKey,
            observe: () => RM.create<String>("Login"),
            builder: (context, loginButtonTxt) {
              return Tab(
                child: Text(
                  loginButtonTxt.value,
                  style: Theme.of(context).textTheme.display1,
                ),
              );
            },
          ),
          StateBuilder<String>(
            rmKey: singUpRMKey,
            observe: () => RM.create<String>("Create new account"),
            builder: (context, singUpTxt) {
              return Tab(
                child: Text(
                  singUpTxt.value,
                  style: Theme.of(context).textTheme.display1,
                ),
              );
            },
          ),
        ],
      ),
    );
  }
}
```

I use TextFormFields autoValidate feature for saving my values to UserVM



```
...  
  final userVMRM = RM.get<UserVM>();
...
...
TextFormField(
                              autovalidate: true,
                              validator: (input) {
                                if (input.length >= 6) {
                                  userVMRM.value.registerLoginPassword = input;
                                  return null;
                                } else {
                                  userVMRM.value.registerLoginPassword = null;
                                  return "Too weak";
                                }
                              },
                            )
```

**Comment by GIfatahTH** [comment]
>Problem is when i was filled email and password forms and switched singup UI then go back to singin uı my UserVM values return to empty.

From the snippet of code you provide, it is more likely that  a new instance of UserVM object is created when you switch between tabs.

To confirm this, add a print to the contractor of UserVM.

**Comment by pMertDogan** [comment]
I found my bug.Your package is working great.

I use same variables for Login & SingUp (  String registerLoginEmail  registerLoginPassword  registerLoginName; )

I use  TextFormField's autoValidate . If my validator logic is false , i was set shared variables to null. I discovered tricky information about autoValidate. When i was switch to SingUp or Login its auto run autoValidate logic, so  when i was swtich to SingUp my variables set to null :)
